### ISSUE TITLE
Screen reader does not announce the name and role of the checkboxes present in the table in windows.

### ISSUE THREAD

**Issue by msftedad** [ceolter/ag-grid#13608]
### Link to reproducible scenario

https://www.ag-grid.com/examples/row-selection/enabling-row-selection/reactFunctionalTs/

### Describe the bug

**Test Environment:**
OS: Windows 11 version 25H2 (OS Build 26200.8246)
Browser: Edge (Version 147.0.3912.60), Chrome (Version 147.0.7727.56)
Screen Reader: NVDA (2025.3.3), JAWS (2026.2603.86), Narrator
URL: https://www.ag-grid.com/examples/row-selection/enabling-row-selection/reactFunctionalTs/

**User Impact:**
Screen reader users are getting impacted because they cannot hear the name and role of the checkboxes in the table. This makes it difficult for them to understand and correctly use the options, leading to confusion and possible mistakes.

**Pre-Requisite:** 
Turn on Screen Readers:
NVDA: Ctrl + Alt + N 
JAWS: Ctrl + Alt + J 
Narrator: Ctrl + win + Enter 
Verbosity:
NVDA: Default 
JAWS: Beginner, Highest 
Narrator: Default

**Repro Steps:**

1. Open the URL: https://www.ag-grid.com/examples/row-selection/enabling-row-selection/reactFunctionalTs/ in edge/ chrome browser.
2. A table will appear on the page.
3. Press tab key to navigate to the check boxes present in the table.
4. Verify whether screen reader announces the name and role of the checkboxes present in the table in windows or not.

**Actual Result:**
Screen reader does not announce the name and role of the checkboxes present in the table in windows.
Screen Reader announces: Press space to toggle row selection (unchecked) row 6 column 1 Press space to select this row

**Refer Attachment:**

1. Screen reader does not announce the name and role of the checkboxes present in the table in windows.png
2. Screen reader does not announce the name and role of the checkboxes present in the table in windows.mp4

https://github.com/user-attachments/assets/eb1c3096-3087-412a-b4c6-2cef22e50891
<img width="1920" height="1266" alt="Image" src="https://github.com/user-attachments/assets/4d239bb5-9f32-4a00-b89e-b2b7deaf35b5" />

**Expected Result:**
Screen reader should announce the name and role of the checkboxes present in the table in windows.
Screen Reader should announce: Press space to toggle row selection Ryan Lochte checkbox (unchecked) row 6 column 1 Press space to select this row

### Version

34.2.0

### Does the issue occur for a specific framework only?

All languages

### Is the issue only observable on a specific browser?

All browsers
